### PR TITLE
Application Insights: obfuscate UUID's in url's which are not static files

### PR DESCRIPTION
**Pull request by filipczyks**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-3462

### Change description
Obfuscating UUID's in urls to {uuid} before being sent to App Insights

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [ ] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/modules/app-insights/index.ts [hunk 1]
```diff
@@ -6,13 +6,37 @@ declare class AppInsightsConfiguration {
   roleName: string
 }
 
+function hideUuidInUrlIfNotStaticFile (url: string): string {
+  let fileRegexp = new RegExp('(\\..{2,5}$)')
```
Comment (timja-kainos) [inline src/main/modules/app-insights/index.ts]: what's this regexp for?

File: src/main/modules/app-insights/index.ts [hunk 2]
```diff
@@ -6,13 +6,37 @@ declare class AppInsightsConfiguration {
...
+  let uuidRegexp = new RegExp('[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89aAbB][a-f0-9]{3}-[a-f0-9]{12}')
```
Comment (timja-kainos) [inline src/main/modules/app-insights/index.ts]: this regex could be constructed once outside of the function and it should be a `const` not `let`

File: src/main/modules/app-insights/index.ts [hunk 3]
```diff
@@ -6,13 +6,37 @@ declare class AppInsightsConfiguration {
...
+
+  if (url.match(fileRegexp) == null) {
```
Comment (timja-kainos) [inline src/main/modules/app-insights/index.ts]: I think .test is more appropriate here:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test

File: src/main/modules/app-insights/index.ts (hunk 1 above)
Comment (filipczyks) [reply src/main/modules/app-insights/index.ts]: Checking if URL is static file, ending with dot and 2 to 5 letters extension. So it prevents us not to rename files.

Comment (timja-kainos) [reply src/main/modules/app-insights/index.ts]: O

**Pull request merged**